src/bin/pgcopydb/ld_transform.c

```diff
@@ -1169,6 +1169,64 @@ parseMessage(StreamContext *privateContext, char *message, JSON_Value *json)
 }
 
 
+/* append the new insert statement if targets same table as previous */
+static bool
+coalesceInsertStatements(LogicalMessageInsert *last,
+						 LogicalMessageInsert *stmt)
+{
+	if (strcmp(last->nspname, stmt->nspname) != 0 ||
+		strcmp(last->relname, stmt->relname) != 0)
+	{
+		return false;
+	}
+
+	LogicalMessageValuesArray *lastValuesArray = &last->new.array->values;
+
+	if (lastValuesArray->capacity < (lastValuesArray->count + 1))
+	{
+		/* double the capacity to avoid excessive realloc */
+		lastValuesArray->capacity *= 2;
+		lastValuesArray->array = (LogicalMessageValues *) realloc(lastValuesArray->array,
+																  sizeof(
+																	  LogicalMessageValues)
+																  *
+																  lastValuesArray->
+																  capacity);
+	}
+
+	/* move the new value to the existing statement */
+	lastValuesArray->array[lastValuesArray->count++] = stmt->new.array->values.array[0];
+
+	/* avoid deallocation of moved value */
+	stmt->new.array->values.count = 0;
+
+	return true;
+}
+
+
+/* Append the new statement if the action is same and targets same table. */
```

Comment:
Formatting: see other function level comments (all are multi-line).
Semantics: needs to also be the same target list.